test/integration/controlplane/transformation/kmsv2_transformation_test.go

```diff
@@ -605,9 +606,12 @@ resources:
 	}
 
 	// Invalidate the DEK by moving the current time forward
-	origNowFunc := kmsv2.NowFunc
-	t.Cleanup(func() { kmsv2.NowFunc = origNowFunc })
-	kmsv2.NowFunc = func() time.Time { return origNowFunc().Add(5 * time.Minute) }
+	origNowFunc := kmsv2.GetNowFunc(kmsName)
+	// Need this cleanup in case of an early failure
+	t.Cleanup(func() { kmsv2.SetNowFuncForTests(kmsName, origNowFunc) })
```

Comment:
Can you just wrap the `SetNowFuncForTests` on the next line instead of doing it twice?

Reply:
updated!